### DISCUSSION TITLE
Multiple many to many relationships across multiple pivot tables

### DISCUSSION THREAD

**Discussion by idealerror** [staudenmeir/eloquent-has-many-deep#188] [Q&A] [answered]
Hello,

I'm struggling with the relationships in my tables.  Looking at the structure in the image below (exclude the single_pivot_table, I scrapped that idea).  A menu item can be on multiple menus, a menu can be on multiple periods, and menus can exist across multiple locations.  All of this is defined and working with data, but my problem is calling a relationship across different relationships. 

Example, I can do `$item->menus` because menus() is a relation of item, but I'm hoping to be able to call `$item->locations` to see the locations that a single item exists within.   

The only way to do this today is to call `Item::with(['menus','menus.periods','menus.periods.locations'])` and then loop over menus, periods, and then locations. 

Any thoughts?  Can this package solve this use case or am I completely crazy with this structure. 

<img width="879" alt="CleanShot 2023-03-21 at 19 58 00@2x" src="https://user-images.githubusercontent.com/13846420/226815891-2c5a6cbf-6b78-49cb-92c0-829a507b8d8e.png">

**Comment by staudenmeir** [comment, marked as answer]
Hi @idealerror,
Define the relationship like this:

```php
class Item extends Model
{
    use \Staudenmeir\EloquentHasManyDeep\HasRelationships;

    public function locations()
    {
        return $this->hasManyDeep(
            Location::class,
            ['menu_items', Menu::class, 'period_menus', Period::class, 'location_periods']
        );
    }
}
```

**Reply by idealerror** [reply]
Amazing, this just worked.  Thanks for the quick reply.  

I've been hitting walls for the past few hours. I'm glad I found this package.
